internal/cloud/backend.go

```diff
@@ -905,10 +905,14 @@ func (b *Cloud) Operation(ctx context.Context, op *backendrun.Operation) (*backe
 	op.Workspace = w.Name
 
 	// Determine the function to call for our operation
-	var f func(context.Context, context.Context, *backendrun.Operation, *tfe.Workspace) (*tfe.Run, error)
+	var f func(context.Context, context.Context, *backendrun.Operation, *tfe.Workspace) (OperationResult, error)
 	switch op.Type {
 	case backendrun.OperationTypePlan:
-		f = b.opPlan
+		if op.Query {
```

Comment:
I suppose it's important to [keep this type as OperationTypePlan](https://github.com/hashicorp/terraform/blob/main/internal/command/query.go#L186) for internal reasons?